Riot/Modules/Room/RoomViewController.m

```diff
@@ -6514,4 +6545,22 @@ - (void)spaceDetailPresenter:(SpaceDetailPresenter *)presenter didJoinSpaceWithI
     [[LegacyAppDelegate theDelegate] openSpaceWithId:spaceId];
 }
 
+#pragma mark - UserSuggestionCoordinatorBridgeDelegate
+
+- (void)userSuggestionCoordinatorBridge:(UserSuggestionCoordinatorBridge *)coordinator
+             didRequestMentionForMember:(MXRoomMember *)member
+                            textTrigger:(NSString *)textTrigger
+{
+    if (textTrigger.length) {
+        NSString *textMessage = [self.inputToolbarView textMessage];
+        textMessage = [textMessage stringByReplacingOccurrencesOfString:textTrigger
```

Comment:
Yes, it can very well do but only if:
* the user suggestion service logic is wrong
* there's a suggestion list popping up
* you tap on one of the suggestions

Honestly I see no way to prevent that apart from making sure the UserSuggestionServiceTests are good. I added an extra check for emails.